### PR TITLE
ref(ui): Convert dropdownAutocomplete to typescript

### PR DESCRIPTION
**Pull request by priscilawebdev**
Convert dropdownAutocomplete to typescript 

#sync-getsentry

### CODE REVIEW

File: src/sentry/static/sentry/app/components/dropdownAutoComplete/types.tsx [hunk 1]
```diff
@@ -0,0 +1,3 @@
+export type Item = any;
```
Comment (priscilawebdev) [inline src/sentry/static/sentry/app/components/dropdownAutoComplete/types.tsx:2]: I will add this type in a follow-up PR.

File: src/sentry/static/sentry/app/components/dropdownAutoComplete/autoCompleteFilter.jsx [hunk 2]
```diff
@@ -0,0 +1,50 @@
+import flatMap from 'lodash/flatMap';
+
+function filterItems(items, inputValue) {
```
Comment (priscilawebdev) [inline src/sentry/static/sentry/app/components/dropdownAutoComplete/autoCompleteFilter.jsx:3]: I will convert these functions in a follow-up PR

File: src/sentry/static/sentry/app/components/assigneeSelector.tsx [hunk 3]
```diff
@@ -246,7 +246,6 @@ const AssigneeSelectorComponent = createReactClass<Props, State>({
         {!loading && (
           <DropdownAutoComplete
             maxHeight={400}
-            zIndex={2}
```
Comment (priscilawebdev) [inline src/sentry/static/sentry/app/components/assigneeSelector.tsx:249]: it looks like that this zIndex was not being used anywhere

**Comment by priscilawebdev** [comment]
I've to check 1 test in jest

File: src/sentry/static/sentry/app/components/dropdownAutoComplete/menu.tsx [hunk 4]
```diff
@@ -0,0 +1,457 @@
...
+  /**
+   * Callback for when dropdown menu closes
+   */
+  onClose?: () => void;
+
+  /**
+   * When AutoComplete input changes
+   */
+  onChange?: (event: React.ChangeEvent<HTMLInputElement>) => void;
+  /**
```
Comment (markstory) [inline src/sentry/static/sentry/app/components/dropdownAutoComplete/menu.tsx:158]:
```suggestion

  /**
```

File: src/sentry/static/sentry/app/components/dropdownAutoComplete/menu.tsx [hunk 5]
```diff
@@ -0,0 +1,457 @@
...
+                  <ItemList data-test-id="autocomplete-list" maxHeight={maxHeight}>
+                    {showNoItems && <EmptyMessage>{emptyMessage}</EmptyMessage>}
+                    {showNoResultsMessage && (
+                      <EmptyMessage>
+                        {noResultsMessage ?? `${emptyMessage} ${t('found')}`}
+                      </EmptyMessage>
+                    )}
+                    {busy && (
+                      <BusyMessage>
+                        <EmptyMessage>{t('Searching...')}</EmptyMessage>
```
Comment (markstory) [inline src/sentry/static/sentry/app/components/dropdownAutoComplete/menu.tsx:348]:
```suggestion
                        <EmptyMessage>{t('Searching\u2026')}</EmptyMessage>
```
Ellipses look better than three .

File: src/sentry/static/sentry/app/components/dropdownAutoComplete/index.tsx [hunk 6]
```diff
@@ -0,0 +1,41 @@
+import React from 'react';
+import styled from '@emotion/styled';
+
+import Menu from 'app/components/dropdownAutoComplete/menu';
```
[the remark this answers is not in the document]
Comment (priscilawebdev) [reply src/sentry/static/sentry/app/components/dropdownAutoComplete/index.tsx:4]: yes..I agree with you, but I was getting some weird error..let me double-check it

**Review by markstory — APPROVED** [review]

**Pull request merged**